src/Lecture2.hs

```diff
@@ -70,7 +75,13 @@ return the removed element.
 >>> removeAt 10 [1 .. 5]
 (Nothing,[1,2,3,4,5])
 -}
-removeAt = error "TODO"
+removeAt :: Int -> [a] -> (Maybe a, [a])
+removeAt i l = go before after $ max i (-1) -- detect negative numbers
```

Comment:
You can check if the number is negative in the beginning of this function so you don't need to handle this case later with pattern matching
```suggestion
removeAt i l
    | i < 0 = ...
    | otherwise = go before after i
```